### PR TITLE
Creates a basic modal component w/ help button

### PR DESCRIPTION
**Pull request by talios0**
No styling yet or a way to pull specific help info from a file.

Creates a button that can be pressed to open a modal menu. Since there's no styling, this appears with the button. When opened, there is a new button that lets you close the window.

### CODE REVIEW

**Comment by Apexal** [comment]
Looks great, though I think it'd be best if we stick to either functional React components or class components. Do you have qualms about switching these to functional components?

File: src/components/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <div>
+        <div className = {modalClass}>
+            <div className = "modal-background"></div>
+            <div className = "modal-card">
+                <header className = "modal-card-head has-text-centered">
+                    <h1 className = "modal-card-title">{props.heading}</h1>
+                    <button className = "delete" onClick = {() => {props.setShow(false);}}></button>
+                </header>
+                <div className = "modal-card-body">
+                    {props.body}
```
Comment (Apexal) [inline src/components/Modal.tsx:25]: Here you could use `props.children` which lets you do something like:
```js
<Modal heading="Hello World">
  content here
</Modal>
```

You'll have to switch `body` in the interface to `JSX.Element[]`

**Review by Apexal — CHANGES_REQUESTED** [review]
Looks good! Check out the small `props.children` comment as it's good to go

File: src/components/Modal.tsx (hunk 1 above)
Comment (talios0) [reply src/components/Modal.tsx:25]: Yeah, that would keep it much cleaner and easier to use! Making the necessary changes now...

**Review by Apexal — APPROVED** [review]
Great! Thanks for making those changes.

**Pull request merged**